File: src/MicroServicioQRs/Dtos/UsuarioQrDto.cs
namespace MicroServicioQRs.Dtos
{
    // Este es EXACTAMENTE el JSON que viaja dentro del QR.
    // GRD3 (app del guarda) escanea, lee este JSON, consulta el usuario
    // por su llave primaria (UsuarioID) y compara dato por dato contra la BD.
    // Elegimos solo lo identificador: robusto y sin JOINs.
    public class UsuarioQrDto
    {
        public Guid UsuarioID { get; set; }
        public string Identificacion { get; set; }
        public string NombreCompleto { get; set; }
    }
}

File: src/app_grd/lib/models/usuario_qr_dto.dart
class UsuarioQrDto {
  final String usuarioID;
  final String identificacion;
  final String nombreCompleto;

  UsuarioQrDto({
    required this.usuarioID,
    required this.identificacion,
    required this.nombreCompleto,
  });

  factory UsuarioQrDto.fromJson(Map<String, dynamic> json) {
  return UsuarioQrDto(
    usuarioID: (json['UsuarioID'] ?? json['usuarioID']).toString(),
    identificacion: (json['Identificacion'] ?? json['identificacion']) as String,
    nombreCompleto: (json['NombreCompleto'] ?? json['nombreCompleto']) as String,
  );
}

  Map<String, dynamic> toJson() {
    return {
      'usuarioID': usuarioID,
      'identificacion': identificacion,
      'nombreCompleto': nombreCompleto,
    };
  }
}